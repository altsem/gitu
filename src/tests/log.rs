use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    commit(ctx.dir.path(), "third commit", "");
    commit(ctx.dir.path(), "second commit", "");
    commit(ctx.dir.path(), "first commit", "");
    ctx
}

#[test]
fn limit_prompt() {
    snapshot!(setup(), "l-n-n");
}

#[test]
fn limit_set_10() {
    snapshot!(setup(), "l-n-n10<enter>");
}

#[test]
fn limit_invalid() {
    snapshot!(setup(), "l-n-nfff<enter>");
}

#[test]
fn limit_2_commits() {
    snapshot!(setup(), "l-n-n2<enter>l");
}

#[test]
fn limit_2_commits_other() {
    snapshot!(setup(), "l-n-n2<enter>l");
}

#[test]
fn grep_prompt() {
    snapshot!(setup(), "l-F");
}

#[test]
fn grep_set_example() {
    snapshot!(setup(), "l-Fexample<enter>");
}

#[test]
fn grep_second() {
    snapshot!(setup(), "l-Fsecond<enter>l");
}

#[test]
fn grep_second_other() {
    snapshot!(setup(), "l-Fsecond<enter>omain<enter>");
}

#[test]
fn log_other_prompt() {
    snapshot!(setup(), "lljlo");
}

#[test]
fn log_other() {
    snapshot!(setup(), "lljlo<enter>");
}

#[test]
fn log_other_input() {
    snapshot!(setup(), "lomain~1<enter>");
}

#[test]
fn log_other_invalid() {
    snapshot!(setup(), "lo <enter>");
}
