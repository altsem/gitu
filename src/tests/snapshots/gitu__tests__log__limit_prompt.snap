---
source: src/tests/log.rs
expression: ctx.redact_buffer()
---
▌On branch main                                                                 |
▌Your branch is ahead of 'origin/main' by 3 commit.                             |
                                                                                |
 Recent commits                                                                 |
 8bb5532 main add first commit                                                  |
 6c08cf7 add second commit                                                      |
 79e63f1 add third commit                                                       |
 b66a0bf origin/main add initial-file                                           |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
? Limit number of commits (default 256): ›                                      |
────────────────────────────────────────────────────────────────────────────────|
Log                     Arguments                                               |
l current               -F Search messages (--grep)                             |
o other                 -n Limit number of commits (-n)                         |
q/<esc> Quit/Close                                                              |
styles_hash: 40d2a2b4b548faf4
